### DISCUSSION TITLE
Not working ater inverter restart. Debugging options?

### DISCUSSION THREAD

**Discussion by kdw2060** [WillCodeForCats/solaredge-modbus-multi#206] [General]
Hi, after my inverter had to be restarted the integration cannot communicate with the inverter anymore.
I already redid the setup in the inverter firmware regarding the activation of modbus protocol and i removed, reinstalled and reconfigured the integration.

Home Assistant logs show this:

```
Logger: pymodbus.client.tcp
Source: /usr/local/lib/python3.10/site-packages/pymodbus/client/tcp.py:187
First occurred: November 29, 2022 at 8:05:47 PM (665 occurrences)
Last logged: 10:59:54 AM

Connection to (192.168.1.57, 1502) failed: [Errno 111] Connection refused
Connection to (192.168.1.57, 1502) failed: timed out
```

Is there a way I can debug the connection myself somehow to find out why the connection is refused?
I'm not familiar with ModBus-TCP, but thinking about a cli program or an api test software like postman that will allow me to send requests from my pc?

**Comment by mpredfearn** [comment]
Connection refused indicates a pretty basic issue - either you've got the wrong port number (have you tried 1502 / 502?) or the inverter simply isn't listening for modbus TCP any more. Has your inverters firmware changed recently as it is a known issue that Solaredge disable modbus TCP over WiFi in certain firmwares and you might have to contact them and ask them to change the firmware to allow it again?
There's not much you will be able to do with your PC other than test if you are able to open a connection to the modbus TCP port (1502 / 502) on the inverter, since that is the first step that is filing, but that is unlikely to be any different to the issue you see with the integration.
